internal/transform/preprocess.go

```diff
@@ -0,0 +1,31 @@
+package transform
+
+import (
+	"fmt"
+	"syscall/js"
+
+	astro "github.com/snowpackjs/astro/internal"
+)
+
+func Preprocess(n *astro.Node) {
+	globalThis := js.Global()
+	stylePreprocess := globalThis.Get("__astro_stylePreprocess")
+	if stylePreprocess.IsUndefined() == true {
+		return
+	}
+
+	// var newValue js.Value
+
+	dataOrPromise := stylePreprocess.Invoke(n.FirstChild.Data, getQuotedAttr(n, "lang"))
+	str := dataOrPromise.String()
+
+	fmt.Println(str)
+	// if str != "" {
+	// 	res, _ := wasm_utils.Await(dataOrPromise)
+	// 	newValue = res[0]
+	// } else {
+	// 	newValue = dataOrPromise
+	// }
```

Comment:
This is commented out, but if the `stylePreprocess` function is `async` we need to block Go until it resolves. 
`wasm_utils.Await(dataOrPromise)` mostly works, but the top-level function needs to wait for it as well. I assume WaitGroups will come in handy here.